meson.build

```diff
@@ -248,9 +261,12 @@ if not headers_only
 				],
 				cpp_args: ['-DFRIGG_HAVE_LIBC'],
 				include_directories: libc_include_dirs,
-				dependencies: libc_deps,
+				# Link with libgcc to get math symbols ...
+				dependencies: [libc_deps, libgcc],
 				link_with: [ldso_lib],
 				link_whole: [libc_sublibs, ldso_lib],
+				# ... but don't export them ourselves
+				link_args: ['-Wl,--exclude-libs=gcc'],
```

Comment:
AFAIU `--exclude-libs` does not work in the static case (ld is not even called here). However, that should not be an issue since the final executable will link against libgcc anyway.